agent/agent-bootstrap/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientMetrics.java

```diff
@@ -0,0 +1,276 @@
+ * client metrics</a>.
+ */
+public final class HttpClientMetrics implements OperationListener {
+
+  private static final double NANOS_PER_MS = TimeUnit.MILLISECONDS.toNanos(1);
+
+  private static final ContextKey<State> HTTP_CLIENT_REQUEST_METRICS_STATE =
+      ContextKey.named("http-client-request-metrics-state");
+
+  private static final Logger logger = Logger.getLogger(HttpClientMetrics.class.getName());
+
+  /**
+   * Returns a {@link OperationMetrics} which can be used to enable recording of {@link
+   * HttpClientMetrics} on an {@link
+   * io.opentelemetry.instrumentation.api.instrumenter.InstrumenterBuilder}.
+   */
+  public static OperationMetrics get() {
+    return HttpClientMetrics::new;
+  }
+
+  private final DoubleHistogram duration;
+  private final LongHistogram requestSize;
+  private final LongHistogram responseSize;
+
+  private HttpClientMetrics(Meter meter) {
+    duration =
+        meter
+            .histogramBuilder("http.client.duration")
+            .setUnit("ms")
+            .setDescription("The duration of the outbound HTTP request")
+            .build();
+    requestSize =
+        meter
+            .histogramBuilder("http.client.request.size")
+            .setUnit("By")
+            .setDescription("The size of HTTP request messages")
+            .ofLongs()
+            .build();
+    responseSize =
+        meter
+            .histogramBuilder("http.client.response.size")
+            .setUnit("By")
+            .setDescription("The size of HTTP response messages")
+            .ofLongs()
+            .build();
+  }
+
+  @Override
+  public Context onStart(Context context, Attributes startAttributes, long startNanos) {
+    return context.with(
+        HTTP_CLIENT_REQUEST_METRICS_STATE,
+        new AutoValue_HttpClientMetrics_State(startAttributes, startNanos));
+  }
+
+  @Override
+  public void onEnd(Context context, Attributes endAttributes, long endNanos) {
+    State state = context.get(HTTP_CLIENT_REQUEST_METRICS_STATE);
+    if (state == null) {
+      logger.log(
+          FINE,
+          "No state present when ending context {0}. Cannot record HTTP request metrics.",
+          context);
+      return;
+    }
+    Attributes durationAndSizeAttributes =
+        applyClientDurationAndSizeView(state.startAttributes(), endAttributes);
+
+    // START APPLICATION INSIGHTS CODE
+
+    // this is needed for detecting telemetry signals that will trigger pre-aggregated metrics via
+    // auto instrumentations
+    Span.fromContext(context).setAttribute(IS_PRE_AGGREGATED, true);
+
+    Attributes durationAttributes =
+        durationAndSizeAttributes.toBuilder()
+            .put(IS_SYNTHETIC, isUserAgentBot(endAttributes, state.startAttributes()))
+            .put(TARGET, getTargetForHttpClientSpan(durationAndSizeAttributes))
+            .build();
+    // END APPLICATION INSIGHTS CODE
```

Comment:
just a preference to make the START/END stand out even more

```suggestion

    // END APPLICATION INSIGHTS CODE
```